java/src/main/java/ai/rapids/cudf/Rmm.java

```diff
@@ -136,6 +136,52 @@ public static boolean isInitialized() throws RmmException {
    */
   public static native long getTotalBytesAllocated();
 
+  /**
+   * Returns the maximum amount of RMM memory (Bytes) outstanding during the
+   * lifetime of the process.
+   *
+   * Note that this result is meaningful when a single thread is using the GPU, or
+   * when we have joined all threads and CUDA synchronized with all streams.
+   */
+  public static native long getMaximumOutstanding();
+
+  /**
+   * Resets a local maximum counter of RMM memory used to keep track of usage between
+   * sections code while debugging.
+   *
+   * @param initialValue an initial value (in Bytes) to use for this local counter
+   */
+  public static void resetLocalMaximumOutstanding(long initialValue) {
```

Comment:
Related to the previous comment, would prefer we use `BytesAllocated` rather than `Outstanding` throughout to make this crystal clear what we're talking about and what units are returned.